src/features/metadata-instance-editor/__tests__/Instance.test.js

```diff
@@ -147,82 +148,149 @@ const userDefinedTemplateField = [
+                    shouldShowCascadingOptions
+                    onModification={jest.fn()}
+                    onRemove={jest.fn()}
+                    onSave={jest.fn()}
+                    canEdit
+                    template={{
+                        fields,
+                    }}
+                />,
+            );
+
+            const editButton = screen.queryByLabelText('Edit Metadata');
+            if (editButton) {
+                fireEvent.click(editButton);
+            }
+            expect(container).toMatchSnapshot();
+        },
+    );
 
-    test('should correctly render templated metadata instance with CascadePolicy', () => {
+    test('should correctly render custom metadata instance', () => {
         const wrapper = shallow(
             <Instance
-                cascadePolicy={{
-                    canEdit: true,
-                    id: 'hello',
-                }}
                 data={data}
                 dataValue="value"
                 intl={intl}
-                isCascadingPolicyApplicable
-                shouldShowCascadingOptions
                 template={{
                     fields,
+                    templateKey: 'properties',
                 }}
             />,
         );
         expect(wrapper).toMatchSnapshot();
     });
 
-    test('should correctly render custom metadata instance', () => {
+    test('should not render footer when not editing', () => {
         const wrapper = shallow(
             <Instance
+                canEdit
                 data={data}
                 dataValue="value"
                 intl={intl}
+                onRemove={jest.fn()}
                 template={{
                     fields,
-                    templateKey: 'properties',
                 }}
             />,
         );
+
         expect(wrapper).toMatchSnapshot();
     });
 
-    test('should not render footer when not editing', () => {
+    test('should correctly render the footer when editing and no cascade policy', () => {
         const wrapper = shallow(
             <Instance
                 canEdit
                 data={data}
                 dataValue="value"
                 intl={intl}
+                onModification={jest.fn()}
                 onRemove={jest.fn()}
+                onSave={jest.fn()}
                 template={{
                     fields,
                 }}
             />,
         );
 
+        wrapper.setState({ isEditing: true });
         expect(wrapper).toMatchSnapshot();
     });
 
-    test('should correctly render the footer', () => {
+    test('should correctly render the footer when editing and cascade policy', () => {
         const wrapper = shallow(
             <Instance
                 canEdit
                 data={data}
                 dataValue="value"
                 intl={intl}
+                cascadePolicy={{
+                    id: 'hello',
+                }}
+                isCascadingPolicyApplicable
```

Comment:
i see `isFolderInstance` is not added here, does the current implementation allow for the `isCascadingPolicyApplicable` and `isFolderInstance` props to work independently of each other? is that okay?

Reply:
They're testing that the footer renders properly. Whether or not the footer renders doesn't depend on these variables.